python/cudf/cudf/core/udf/groupby_utils.py

```diff
@@ -154,7 +154,10 @@ def jit_groupby_apply(offsets, grouped_values, function, *args):
     offsets = cp.asarray(offsets)
     ngroups = len(offsets) - 1
 
-    output = cudf.core.column.column_empty(ngroups, dtype=return_type)
+    # numba doesn't support masks
+    output = cudf.core.column.column_empty(
```

Comment:
This feels like we're losing performance by allocating a null mask with all nulls, and then dropping it. Is there a better way?

Reply:
Fair point. I added a `for_numba` argument in this function to avoid creating a mask in these cases